### PR TITLE
Fix/web links

### LINKED ISSUE #2262: Links usable on the web
A great improvement to the web application would be proper integration of the URLs in the browser
Currently, we don't show the current page link in the address bar and opening a link to a specific page returns to the default page. 

I think the best way to do this without affecting the mobile part is to use deep links from react-navigation 6
https://reactnavigation.org/docs/configuring-links

As the current mobile behavior, it should be possible to go back and forward using the browser back/forward buttons

### PR DESCRIPTION
**Pull request by frankcalise**
## Please verify the following:

- [x] `yarn test` **jest** tests pass with new tests, if relevant

## Describe your PR
- Closes #2262
- Adds web link support according to [react-navigation docs](https://reactnavigation.org/docs/configuring-links)
- Configures `DemoShowroomScreen` to have slugified query params for section / item navigation, ie: `/showroom/text/presets`
- Adds in example of Platform.select() specific ListItem component under `DemoShowroomScreen`
- Patches Expo web dev server issue with `package.json` resolutions (see https://github.com/expo/expo/issues/19833) for details

image

### CODE REVIEW

**Comment by frankcalise** [comment]
@Dolabok would like your feedback here too!

File: boilerplate/package.json [hunk 1]
```diff
@@ -125,7 +126,8 @@
   },
   "resolutions": {
     "@types/react": "^17",
-    "@types/react-dom": "^17"
+    "@types/react-dom": "^17",
+    "@expo/dev-server": "0.1.120"
```
Comment (frankcalise) [inline boilerplate/package.json:130]: Currently an issue with Expo cli https://github.com/expo/expo/issues/19833

File: boilerplate/app.json [hunk 2]
```diff
@@ -4,6 +4,7 @@
   "expo": {
     "name": "HelloWorld",
     "slug": "HelloWorld",
+    "scheme": "helloworld",
```
Comment (frankcalise) [inline boilerplate/app.json:7]: Added placeholder that `new` command will replace, necessary for deep links

File: boilerplate/app/screens/DemoShowroomScreen/DemoShowroomScreen.tsx [hunk 3]
```diff
@@ -18,143 +29,206 @@ export interface Demo {
...
+    const timeout = useRef<ReturnType<typeof setTimeout>>()
+    const drawerRef = useRef<DrawerLayout>()
+    const listRef = useRef<SectionList>()
+    const menuRef = useRef<FlatList>()
+    const progress = useSharedValue(0)
+    const route = useRoute<RouteProp<DemoTabParamList, "DemoShowroom">>()
+    const params = route.params
+
+    // handle Web links
+    React.useEffect(() => {
```
Comment (frankcalise) [inline boilerplate/app/screens/DemoShowroomScreen/DemoShowroomScreen.tsx:99]: Translates slugs into section and item index to utilize existing handleScroll

File: boilerplate/app/screens/DemoShowroomScreen/DemoShowroomScreen.tsx [hunk 4]
```diff
@@ -18,143 +29,206 @@ export interface Demo {
...
+
+const slugify = (str) =>
+  str
+    .toLowerCase()
+    .trim()
+    .replace(/[^\w\s-]/g, "")
+    .replace(/[\s_-]+/g, "-")
+    .replace(/^-+|-+$/g, "")
+
+const WebLisItem: FC<DemoListItem> = ({ item, sectionIndex }) => {
```
Comment (yulolimum) [inline boilerplate/app/screens/DemoShowroomScreen/DemoShowroomScreen.tsx:46]:
```suggestion
const WebListItem: FC<DemoListItem> = ({ item, sectionIndex }) => {
```

File: boilerplate/app/screens/DemoShowroomScreen/DemoShowroomScreen.tsx [hunk 5]
```diff
@@ -18,143 +29,206 @@ export interface Demo {
...
+          onPress={() => handleScroll(sectionIndex, index + 1)}
+          text={u}
+          rightIcon={isRTL ? "caretLeft" : "caretRight"}
+        />
+      ))}
+    </View>
+  )
+}
+
+const ShowroomListItem = Platform.select({ web: WebLisItem, default: NativeListItem })
```
Comment (yulolimum) [inline boilerplate/app/screens/DemoShowroomScreen/DemoShowroomScreen.tsx:85]:
```suggestion
const ShowroomListItem = Platform.select({ web: WebListItem, default: NativeListItem })
```

**Review by yulolimum — APPROVED** [review]
Great work!

**Comment by Dolabok** [comment]
Great PR !

**Pull request merged**